### ISSUE TITLE
Tag regex too strict in pydantic schema

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#17999]
from https://sentry.galaxyproject.org/share/issue/b9279b5b737f4d938b0961f103461e6d/:

```
36 validation errors for HistoryContentsResult
2.HDACustom.tags.0
  String should match pattern '^([^\s.:])+(.[^\s.:]+)*(:[^\s.:]+)?$' [type=string_pattern_mismatch, input_value='group:S.', input_type=str]
    For further information visit https://errors.pydantic.dev/2.5/v/string_pattern_mismatch
2.HDADetailed.tags.0
  String should match pattern '^([^\s.:])+(.[^\s.:]+)*(:[^\s.:]+)?$' [type=string_pattern_mismatch, input_value='group:S.', input_type=str]
    For further information visit https://errors.pydantic.dev/2.5/v/string_pattern_mismatch
2.HDADetailed.accessible
  Field required [type=missing, input_value={'id': 127377110, 'state'...toryDatasetAssociation'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
2.HDADetailed.file_ext
  Field required [type=missing, input_value={'id': 127377110, 'state'...toryDatasetAssociation'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
2.HDADetailed.file_size
```

Tag in question is `group:S.'` which looks fine and has passed whatever validation we do internally.

**Comment by jdavcs** [comment]
`group:S.` appears to be not a valid tag. As expected, the regex corresponds to the [client-side regex](https://github.com/galaxyproject/galaxy/blob/release_24.0/client/src/components/Tags/model.js#L10), which validates input as expected: I tried saving a tag in that format in different ways, but client-side validation caught it every time. Yet, the `group:S.` exists in the db, and it is a day old. For now, I have no idea how that got into the system. (the new sentry events are were triggered by me looking at that data on main)

The trailing period in the tag was clearly a typo (based on what the other tags are in that history). Relaxing the regex and allowing trailing `[.:]` is the most straightforward solution here, but only if we really want to allow those trailing characters. Should we allow them?

**Comment by jdavcs** [comment]
Fixed in #18025
